File: lca/src/main/java/com/dreampany/lca/service/NotifyService.java
package com.dreampany.lca.service;

import com.dreampany.frame.api.service.BaseJobService;
import com.dreampany.lca.vm.NotifyViewModel;
import com.firebase.jobdispatcher.JobParameters;

import org.jetbrains.annotations.NotNull;

import javax.inject.Inject;

/**
 * Created by Hawladar Roman on 7/22/2018.
 * Dreampany Ltd
 * dreampanymail@gmail.com
 */
public class NotifyService extends BaseJobService {

    @Inject
    NotifyViewModel vm;

    @Override
    protected boolean doJob(@NotNull JobParameters job) {
        vm.notifyIf();
        return true;
    }

    @Override
    protected boolean done(@NotNull JobParameters job) {
        vm.clear();
        return true;
    }
}

File: history/src/main/kotlin/com/dreampany/history/service/NotifyService.kt
package com.dreampany.history.service

import com.dreampany.history.vm.NotifyViewModel
import com.dreampany.frame.api.service.BaseJobService
import com.firebase.jobdispatcher.JobParameters
import javax.inject.Inject

/**
 * Created by Roman-372 on 7/24/2019
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
class NotifyService : BaseJobService() {

    @Inject
    internal lateinit var vm: NotifyViewModel

    override fun doJob(job: JobParameters): Boolean {
        vm.notifyIf()
        return true
    }

    override fun done(job: JobParameters): Boolean {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }
}